### PR TITLE
Add STIG tags to ansible_tags

### PR DESCRIPTION
**Pull request by shawndwells**

### CODE REVIEW

**Comment by shawndwells** [comment]
marking as WIP. Encountering for-loop errors, will cleanup tomorrow.

**Comment by mpreisler** [comment]
@shawndwells There is a reason for the weird formatting. You need to keep it!

**Comment by shawndwells** [comment]
but now I get a different build error:
`````
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 76 element for-each
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 77 element for-each
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 76 element for-each
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 77 element for-each
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 76 element for-each
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 77 element for-each
The 'select' expression does not evaluate to a node set.
runtime error: file /tmp/scap-security-guide/shared/transforms/xccdf-addremediations.xslt line 76 element for-each
The 'select' expression does not evaluate to a node set.
`````

**Comment by shawndwells** [comment]
Added xsl:if that took care of build errors. Not sure where this is falling apart. How the role looks now:

`````
      tags:
        - ensure_gpgcheck_globally_activated
        - high_severity
        - unknown_strategy
        - low_complexity
        - medium_disruption
        - CCE-26989-4
        - NIST-800-53-CM-5(3)
        - NIST-800-53-SI-7
        - NIST-800-53-MA-1(b)
        - NIST-800-171-3.4.8
        - PCI-DSS-Req-6.2
        - CJIS-5.10.4.1
`````

Guessing the STIG ID URLs are incorrect, but they match how the <ref> tags are setup elsewhere (I think).

File: shared/transforms/xccdf-addremediations.xslt [hunk 1]
```diff
@@ -71,7 +73,9 @@
     - NIST-800-53-<xsl:value-of select="."/></xsl:for-each><xsl:for-each select="$ref_nist800_171">
     - NIST-800-171-<xsl:value-of select="."/></xsl:for-each><xsl:for-each select="$ref_pci_dss">
     - PCI-DSS-<xsl:value-of select="."/></xsl:for-each><xsl:for-each select="$ref_cjis">
-    - CJIS-<xsl:value-of select="."/></xsl:for-each></xsl:variable>
+    - CJIS-<xsl:value-of select="."/></xsl:for-each><xsl:if test="$ref_disa_ossrg"><xsl:for-each select="$ref_disa_ossrg">
```
Comment (mpreisler) [inline shared/transforms/xccdf-addremediations.xslt]: What is the if for? why is it needed for OSSRG and not for other references? shouldn't for-reach over empty node set just not iterate at all?

File: shared/transforms/shared_constants.xslt [hunk 2]
```diff
@@ -13,7 +13,7 @@
 <xsl:variable name="cnss1253uri">http://www.cnss.gov/Assets/pdf/CNSSI-1253.pdf</xsl:variable>
 <xsl:variable name="dcid63uri">not_officially_available</xsl:variable>
 <xsl:variable name="disa-cciuri">http://iase.disa.mil/stigs/cci/Pages/index.aspx</xsl:variable>
-<xsl:variable name="disa-srguri">http://iase.disa.mil/stigs/srgs/Pages/index.aspx</xsl:variable>
+<xsl:variable name="disa-srguri">https://iase.disa.mil/stigs/os/general/Pages/index.aspx/</xsl:variable>
```
Comment (mpreisler) [inline shared/transforms/shared_constants.xslt]: This change breaks OpenSCAP HTML report. I agree we need to do it but it definitely needs to be a separate PR with tracking to the OpenSCAP change. Please don't clump multiple changes into one PR.

Comment (jan-cerny) [reply shared/transforms/shared_constants.xslt]: @mpreisler I think we should think about making OpenSCAP HTML report more clever. This affects us each time when some URL changes.

File: shared/transforms/xccdf-addremediations.xslt (hunk 1 above)
Comment (jan-cerny) [reply shared/transforms/xccdf-addremediations.xslt]: It looks like the `<if>` isn't necessary.

**Comment by shawndwells** [comment]
On 8/30/17 8:40 AM, Jan Černý wrote:
>
> *@jan-cerny* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In shared/transforms/xccdf-addremediations.xslt
> <https://github.com/OpenSCAP/scap-security-guide/pull/2286#discussion_r136054815>:
>
> > @@ -71,7 +73,9 @@
>      - NIST-800-53-<xsl:value-of select="."/></xsl:for-each><xsl:for-each select="$ref_nist800_171">
>      - NIST-800-171-<xsl:value-of select="."/></xsl:for-each><xsl:for-each select="$ref_pci_dss">
>      - PCI-DSS-<xsl:value-of select="."/></xsl:for-each><xsl:for-each select="$ref_cjis">
> -    - CJIS-<xsl:value-of select="."/></xsl:for-each></xsl:variable>
> +    - CJIS-<xsl:value-of select="."/></xsl:for-each><xsl:if test="$ref_disa_ossrg"><xsl:for-each select="$ref_disa_ossrg">
>
> It looks like the |<if>| isn't necessary.
>

Tried it without the xsl:if and the build fails. Not sure why -- it
looks like ref_disa_ossrg is not being populated. Haven't had time to
dive into it (currently end of fiscal quarter).

File: shared/transforms/shared_constants.xslt (hunk 2 above)
Comment (mpreisler) [reply shared/transforms/shared_constants.xslt]: I agree, the report should recognize new and old URLs to be compatible with a wide variety of existing SCAP content.

**Comment by mpreisler** [comment]
Recovered into https://github.com/mpreisler/scap-security-guide/tree/shawn_2286

Will fix it up and open a new PR when I get some extra time.
